adapter/internal/discovery/xds/server.go

```diff
@@ -666,6 +782,137 @@ func deleteAPI(apiIdentifier string, environments []string, organizationID strin
+								newLatestMajorRangeApi.Version,
+								GetMajorMinorVersionRangeRegex(*newLatestMajorRangeApi),
+							)
+							pathSpecifier := &routev3.RouteMatch_SafeRegex{
+								SafeRegex: &envoy_type_matcherv3.RegexMatcher{
+									Regex: regex,
+								},
+							}
+							route.Match.PathSpecifier = pathSpecifier
+							action := &routev3.Route_Route{}
+							action = route.Action.(*routev3.Route_Route)
+							action.Route.RegexRewrite.Pattern.Regex = regexRewritePattern
+							route.Action = action
+						}
+					} else {
+						delete(orgIDLatestAPIVersionMap[organizationID][apiRangeIdentifier], majorVersionRange)
+					}
+				}
+			}
+			minorVersionRange := GetMinorVersionRange(*deletingApiSemVersion)
+			if deletingApisMinorRangeLatestApi, ok := latestAPIVersionMap[minorVersionRange]; ok {
+				if deletingApisMinorRangeLatestApi.Version == api.GetVersion() {
+					newLatestMinorRangeApi := &SemVersion{
+						Version: "",
+						Major:   deletingApiSemVersion.Major,
+						Minor:   deletingApiSemVersion.Minor,
+						Patch:   nil,
+					}
+					newLatestMinorRangeApiUuid := ""
+					for apiUUID, swagger := range orgIDAPIMgwSwaggerMap[organizationID] {
+						// Iterate all the API versions other than the deleting API itself
+						if swagger.GetTitle() == api.GetTitle() && apiUUID != apiIdentifier {
+							currentApiSemVersion, _ := ValidateAndGetVersionComponents(swagger.GetVersion())
+							if currentApiSemVersion != nil {
+								if currentApiSemVersion.Major == deletingApiSemVersion.Major &&
+									currentApiSemVersion.Minor == deletingApiSemVersion.Minor {
+									if CompareSemanticVersions(*newLatestMinorRangeApi, *currentApiSemVersion) {
+										newLatestMinorRangeApi = currentApiSemVersion
+										newLatestMinorRangeApiUuid = apiUUID
+									}
+								}
+							}
+						}
+					}
+					if newLatestMinorRangeApiUuid != "" && newLatestMinorRangeApiUuid != newLatestMajorRangeApiUuid {
+						orgIDLatestAPIVersionMap[organizationID][apiRangeIdentifier][minorVersionRange] = *newLatestMinorRangeApi
+						for _, route := range orgIDOpenAPIRoutesMap[organizationID][newLatestMinorRangeApiUuid] {
+							regex := route.GetMatch().GetSafeRegex().GetRegex()
+							regex = strings.ReplaceAll(
+								regex,
+								newLatestMinorRangeApi.Version,
+								GetMinorVersionRangeRegex(*newLatestMinorRangeApi),
+							)
+							pathSpecifier := &routev3.RouteMatch_SafeRegex{
+								SafeRegex: &envoy_type_matcherv3.RegexMatcher{
+									Regex: regex,
+								},
+							}
+							regexRewritePattern := route.GetRoute().GetRegexRewrite().GetPattern().GetRegex()
+							regexRewritePattern = strings.ReplaceAll(
+								regexRewritePattern,
+								newLatestMinorRangeApi.Version,
+								GetMinorVersionRangeRegex(*newLatestMinorRangeApi),
+							)
+							route.Match.PathSpecifier = pathSpecifier
+							action := &routev3.Route_Route{}
+							action = route.Action.(*routev3.Route_Route)
```

Comment:
Can we improve considering below structure ?
```
action := &routev3.Route_Route{
		Action: &routev3.RouteAction{
....
```

Reply:
This is not possible since the full object construction is not possible using the data we have on the current point of the execution.